geps/gep-1619.md

```diff
@@ -106,10 +200,6 @@ The following are items that we intend to resolve before we consider this GEP im
 - We need to add a small section on compliance regarding the browser and client relationship.
 - We need to design and document an API for session persistence and/or session affinity in the Gateway API spec.
 
-## API
-
-TBD
-
 ## Alternatives
 
 ### Alternate Naming
```

Comment:
My suggestion is to not make a big distinction between persistence and affinity and mention that these 2 terms are used interchangeably or with various meanings in the industry. So we could just use the term session persistence through out to also include the session affinity (as described here).

One more aspect to be explored: what happens when the affinity/persistence is broken because the given backend is not up/healthy? If that's an error case how should that be handled? Should the API dictate the http error code? Or should the API dictate fall back behavior? One fallback behavior is that when a backend for a session is not up or healthy the load balancer finds a new backend (using the applicable load balancing policy) and create a new cookie/header thereby creating a new session for this otherwise failing RPC.

Reply:
Discussion for naming is here: https://github.com/kubernetes-sigs/gateway-api/pull/1935/files#r1177654213

Great question. Added as an open question for now to address later, as it seems like an implementation detail we can address in the near future.

Though, I don't fully understand the implications of Gateway API dictating failure modes and edge cases such as this error scenario or even something like connection draining. Is there precedent for this in the API today? If an implementation handled this error case differently, would they be non-compliant? How prescriptive is Gateway API for something like session persistence?

Reply:
> implications of Gateway API dictating failure modes and edge cases...
> How prescriptive is Gateway API for something like session persistence?

Good questions and I guess there are no good answers other than saying "it depends". My thoughts are as follows: to make the API truly portable and useful it will be good to have a tight spec that includes failures modes, fallback behavior and interaction with (on dependency on) other APIs such as the things we have talked about.